test/augmentation/test_augmentation.py

```diff
@@ -24,6 +25,787 @@
+        expected_output_shape = torch.Size((2, 3, 4, 5))
+        test_input = torch.rand((2, 3, 4, 5), device=self.device, dtype=self.dtype)
+        output = augmentation(test_input)
+        assert (output == test_input).all()
+
+    def _test_random_p_0_return_transform_implementation(self, params):
+        augmentation = self._create_augmentation_from_params(**params, p=0.0, return_transform=True)
+        expected_output_shape = torch.Size((2, 3, 4, 5))
+        expected_transformation_shape = torch.Size((2, 3, 3))
+        test_input = torch.rand((2, 3, 4, 5), device=self.device, dtype=self.dtype)
+        output, transformation = augmentation(test_input)
+
+        assert (output == test_input).all()
+        assert transformation.shape == expected_transformation_shape
+        assert (transformation == kornia.eye_like(3, transformation)).all()
+
+    def _test_random_p_1_implementation(self, input_tensor, expected_output, params):
+        augmentation = self._create_augmentation_from_params(**params, p=1.0, return_transform=False)
+        output = augmentation(input_tensor.to(self.device).to(self.dtype))
+
+        # Output should match
+        assert output.shape == expected_output.shape
+        assert_allclose(output, expected_output.to(self.device).to(self.dtype), atol=1e-4, rtol=1e-4)
+
+    def _test_random_p_1_return_transform_implementation(
+            self, input_tensor, expected_output, expected_transformation, params):
+        augmentation = self._create_augmentation_from_params(**params, p=1.0, return_transform=True)
+        output, transformation = augmentation(input_tensor.to(self.device).to(self.dtype))
+        # Output should match
+        assert output.shape == expected_output.shape
+        assert_allclose(output, expected_output.to(self.device).to(self.dtype), atol=1e-4, rtol=1e-4)
+        # Transformation should match
+        assert transformation.shape == expected_transformation.shape
+        assert_allclose(transformation, expected_transformation.to(
+            self.device).to(self.dtype), atol=1e-4, rtol=1e-4)
+
+    def _test_module_implementation(self, params):
+        augmentation = self._create_augmentation_from_params(**params, p=0.5, return_transform=True)
+
+        augmentation_sequence = nn.Sequential(augmentation, augmentation)
+
+        input_tensor = torch.rand(3, 5, 5, device=self.device, dtype=self.dtype)  # 3 x 5 x 5
+
+        torch.manual_seed(42)
+        out1, transform1 = augmentation(input_tensor)
+        out2, transform2 = augmentation(out1)
+        transform = transform2 @ transform1
+
+        torch.manual_seed(42)
+        out_sequence, transform_sequence = augmentation_sequence(input_tensor)
+
+        assert out2.shape == out_sequence.shape
+        assert transform.shape == transform_sequence.shape
+        assert_allclose(out2, out_sequence)
+        assert_allclose(transform, transform_sequence)
+
+    def _test_inverse_coordinate_check_implementation(self, params):
```

Comment:
I think this test is really important since it makes sure, the returned transformation matrix corresponds to the applied transformation. Ofc it could still go wrong but for a sainty check I think it's quite good. I found 4-5 of the bugs fixed in previous PRs with this check.